ivy/functional/ivy/nest.py

```diff
@@ -1055,6 +1055,119 @@ def nested_map(
+
+    With :class:`ivy.Container` input:
+
+    >>> x = ivy.Container(
+        a=ivy.array([[1, 2, 3], [9, 8, 7]]) , b=ivy.array([[4, 5, 6], [12, 13, 14]])
+        )
+    >>> function = lambda a : a  + 1
+    >>> ivy.nested_map(x, function)
+    >>> print(x)
+    {
+        a: ivy.array([[2, 3, 4], [10, 8, 8]]),
+        b: ivy.array([[5, 6,  7], [13, 14, 15]])
+    }
+
+    >>> nest = [[1, 2], (3, 4), {'a': 5, 'b': 6}]
+    >>> ivy.nested_map(lambda x: x * 2, nest, to_ignore=(list, dict))
+    [[2, 4], (6, 8), {'a': 5, 'b': 6}]
+
+    Note:
+        - The function should be a callable that takes one argument.
+        - The iterable should be a nested iterable, such as a list of lists.
+        - The to_ignore parameter allows specifying types to be ignored when deciding
+          whether to go deeper into the nest or not.
+        - By default, no types are ignored, and all nested types are recursively mapped.
+        - Providing types in to_ignore prevents the function from recursively
+          mapping those types and stops the recursion at that level.
+
+    >>> nest = [[1, 2], {'a': [3, 4]}, 'hello']
+    >>> ivy.nested_map(lambda x: x * 2, nest, extra_nest_types=(dict,))
+    [[2, 4], {'a': [6, 8]}, 'hello']
+
+    Note:
+        - The function should be a callable that takes one argument.
+        - The iterable should be a nested iterable, such as a list of lists.
+        - The extra_nest_types parameter allows specifying additional types to
+          recursively check when deciding whether to go deeper into the nest or not.
+        - By default, only the base types (list, tuple, and dict) are
+          recursively checked.
+        - Providing extra types allows customizing the recursive behavior for
+          specific types not handled by default.
+
+    >>> nest = [[1, 2], (3, 4), {'a': 5, 'b': 6}]
+    >>> ivy.nested_map(lambda x: x * 2, nest, include_derived=True)
+    [[2, 4], (6, 8), {'a': 10, 'b': 12}]
+
+    Note:
+        - The function should be a callable that takes one argument.
+        - The iterable should be a nested iterable, such as a list of lists.
+        - If include_derived is True, the mapping is applied not only to the
+          base types (list, tuple, dict) but also to classes derived from them.
+        - By default, the mapping is limited to the base types (list, tuple, dict)
+          and does not apply to derived classes.
+
+    >>> ivy.nested_map(lambda x: x * 2, [[1, 2], [3, 4], [5, 6]], max_depth=1)
+    [[2, 4], [6, 8], [10, 12]]
+
+    Note:
+        - The function should be a callable that takes one argument.
+        - The iterable should be a nested iterable, such as a list of lists.
+        - If max_depth is None, there is no limit on the nested depth.
+        - If max_depth is set to an integer value, the mapping stops when the
+          maximum depth is reached.
+
+    >>> ivy.nested_map(lambda x: x * 2, [[1, 2], [3, 4], [5, 6]], to_mutable=True)
```

Comment:
In the last iteration you where using `Tuple` that worked well with this example, don't you think you should present it instead?